images/gitlab/tests/main.tf

```diff
@@ -0,0 +1,98 @@
+terraform {
+  required_providers {
+    oci = { source = "chainguard-dev/oci" }
+  }
+}
+
+variable "digests" {
+  description = "The image digest to run tests over."
+  type = object({
+    kas = string
+  })
+}
+
+variable "namespace" {
+  description = "The namespace to install Gitlab in."
+}
+
+// Invoke the image's version command.
+// $IMAGE_NAME is populated with the image name by digest.
+// TODO: Update or remove this test as appropriate.
+data "oci_exec_test" "kas-version" {
+  digest = var.digests.kas
+  script = "docker run --rm $IMAGE_NAME -v"
+}
+
+resource "random_pet" "suffix" {}
+
+locals {
+  namespace = "gitlab-${random_pet.suffix.id}"
+}
+
+data "oci_string" "kas-ref" { input = var.digests.kas }
+
+resource "helm_release" "gitlab" {
+  name = "gitlab"
```

Comment:
We should namespace this too

Reply:
but it is namespaced  `namespace        = local.namespace` using a random namespace, as previously defined in the `namespace` variable.

Reply:
Generally we do it to both to avoid conflicts.  I think in most cases things are namespaced by the namespace, so this being fixed matters less, but I'd probably do it just for the consistency with everywhere else.

Reply:
Changing the code to look like cer-manager so this will be different in any case.

Reply:
I reorganized the code following cert-manager images.